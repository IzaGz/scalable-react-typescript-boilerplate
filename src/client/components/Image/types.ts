import * as React from 'react'; 
import Image from './';

export type ImageSize = 'xxsmall' | 'xmall' | 'small' | 'medium' | 'large' | 'xlarge';

export interface ImageProps extends React.Props<typeof Image> {
  size?: ImageSize;
  src: string;
  alt: string;
}
