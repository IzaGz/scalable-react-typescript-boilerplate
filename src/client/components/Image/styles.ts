import remStringFromPx from '../utils';
const styled = require('styled-components').default;

const imageSizeMap = {
  xxsmall: 100,
  xsmall: 200,
  small: 300,
  medium: 400,
  large: 500,
  xlarge: 600,
};

export const Img = styled.img`
  max-width: 100%;
  height: auto;
  max-height: ${(props) => remStringFromPx(imageSizeMap[props.size])};
  display:block
`;
