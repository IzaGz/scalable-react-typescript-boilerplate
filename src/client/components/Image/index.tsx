import * as React from 'react';
import { Img } from './styles';
import { ImageProps } from './types';

export default function Image({
  ...props,
}: ImageProps): JSX.Element {
  return(
    <Img {...props} />
  );
};
