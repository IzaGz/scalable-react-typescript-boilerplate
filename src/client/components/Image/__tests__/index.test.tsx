import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Image from '../';
import { MockImageProps } from './__mocks__/imageMocks.mock';

describe('<Button /> default', () => {
  it('should render a plain button with default props and required props (src, alt)', () => {
    const wrapper = shallow(
      <Image alt="alt text" src="src string" />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render a plain button with other props', () => {
    const wrapper = shallow(
      <Image {...MockImageProps} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
