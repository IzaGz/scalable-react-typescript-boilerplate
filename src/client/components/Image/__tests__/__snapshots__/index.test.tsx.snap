exports[`<Button /> default should render a plain button with default props and required props (src, alt) 1`] = `
<styled.img
  alt="alt text"
  size="small"
  src="src string" />
`;

exports[`<Button /> default should render a plain button with other props 1`] = `
<styled.img
  alt="Some alt text"
  size="medium"
  src="A source string" />
`;
